pkg/tests/tfcheck/tfcheck.go

```diff
@@ -84,6 +101,39 @@ func NewTfDriver(t pulcheck.T, dir, providerName string, prov *schema.Provider)
 	}
 }
 
+// Only exported for use in PF
+func NewTFDriverV6(t pulcheck.T, dir, providerName string, prov providerv6) *TfDriver {
```

Comment:
The signature here is a tad weird, we have

```
NewTFDriverV6
NewTFDriverSDK
```

Would it be possible instead to have

```
NewTFDriver(, <any-TF-provider>)
```

Functionality in https://github.com/hashicorp/terraform-plugin-mux can be used to mix and match and upgrade providers from SDK to protocol V6.

Reply:
The thinking here is that PF also supports V5, so we could potentially also want that.
These are all internal, so let's keep it as is for now and refactor as we see a need. It's not obvious that it's going to see any more use than what we currently have.